### PR TITLE
[max] Incorporated new QuantityType (Units of Measurement)

### LINKED ISSUE #3689: [max!] MAX! Binding keeps discovering Things which are already defined in *.things file
## Expected Behavior
MAX! Binding should ignore "discovered" things which are already defined as things in *.things file

## Current Behavior
MAX keeps showing these things as newly discovered in the Inbox.

## Steps to Reproduce (for Bugs)
1. define a thing in *.thing file
2. restart openhab
3. check Inbox -> Thing pops up

## Your Environment
latest OH2.4.0 on openhabian on RPi3B+

### PR DESCRIPTION
**Pull request by cweitkamp**
- Incorporated new QuantityType (Units of Measurement)
- Replaced `Runnable`
- Added 'representation-property' tag in brige type an thing type definitions
- Removed dependencies for `com.google.common.*` (Guava)

Fixes #3689 

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

**Comment by wborn** [comment]
It's great to see UoM and lambdas on this binding! The code looks good but the documentation also needs to be updated.

It also looks like this PR will cause more merge conflicts on the dormant https://github.com/openhab/openhab2-addons/pull/1789 which also improves the Max binding code.

**Comment by cweitkamp** [comment]
I added the missing documentation.

I subscribed to #1789 to receive an notice once it will have been merged. I can rebase this contribution afterwards.

**Comment by martinvw** [comment]
@cweitkamp can you fix the conflicts here? Thanks!

**Comment by cweitkamp** [comment]
@martinvw I have rebased my branch.

File: addons/binding/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/handler/MaxDevicesHandler.java [hunk 1]
```diff
@@ -340,15 +335,27 @@ public void handleCommand(ChannelUID channelUID, Command command) {
...
-            maxCubeBridge.queueCommand(sendCommand);
-        } else {
-            logger.warn("Setting of channel {} not possible. Read-only", channelUID);
+        switch (channelUID.getId()) {
+            case CHANNEL_SETTEMP:
+                if (refreshingActuals) {
+                    refreshActualsRestore();
+                }
+                Command tempCommand = command;
+                if (command instanceof QuantityType) {
```
Comment (marcelrv) [inline addons/binding/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/handler/MaxDevicesHandler.java]: This breaks the compatibility with OH2.3 see issue #4217
OH2.3 will not send a quanitytype, hence all temp settings are ignored

Comment (cweitkamp) [reply addons/binding/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/handler/MaxDevicesHandler.java]: Yes, it does. But why do you want to use this version with OH2.3?

Side note: These lines of code have been moved to
https://github.com/openhab/openhab2-addons/blob/e0ec54d569a387a238c6dee3b2fbf9bfef3a6881/addons/binding/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/handler/MaxCubeBridgeHandler.java#L419-423

Comment (marcelrv) [reply addons/binding/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/handler/MaxDevicesHandler.java]: if I'm not mistaken, the latest version of the binding is installed.
Hence a OH2.3 (except the release version) will pull the latest one as far as I understand

**Pull request merged**